packages/block-library/src/navigation/edit/index.js

```diff
@@ -529,13 +540,17 @@ function Navigation( {
 										</>
 									) }
 								</Button>
-								{ overlayMenuPreview && (
-									<OverlayMenuPreview
-										setAttributes={ setAttributes }
-										hasIcon={ hasIcon }
-										icon={ icon }
-									/>
-								) }
+								<div id={ overlayMenuPreviewId }>
+									{ overlayMenuPreview && (
```

Comment:
Accessibility question. I couldn't find any definitive best practice for whether the element being targetted by `aria-controls` should always remain in the DOM. 

Currently the _contents_ are added/removed by React on toggle. But should we instead leave it in the DOM and show/hide using _CSS_?

Reply:
I would think so. Here is an example that helps.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls

This uses the tab panel use-case where the tab content is hidden via CSS but not totally removed from the DOM.

Why is this panel being removed from the DOM anyway? Performance reasons?

Reply:
Just tested this and could not find any negative effects. No warnings in console after opening/closing.

Reply:
> Why is this panel being removed from the DOM anyway? Performance reasons?

Generally it seems to be a common pattern in React. I believe it's because it's easier for the developer to simply not-render something based on state rather than figure out the best way to hide via CSS.

Sometimes I guess rendering a component when it's not needed might trigger performance issues so that's worth bearing in mind.

In this case however we should be ok to leave the items in the DOM.